packages/host/app/lib/file-def-manager.ts

```diff
@@ -432,4 +506,49 @@ export default class FileDefManagerImpl implements FileDefManager {
       }
     }
   }
+
+  private async onTimelineEvent(event: MatrixEvent) {
+    const content = event.getContent();
+    if (!content) {
+      return;
+    }
+
+    const eventType = event.getType();
+
+    // Handle attached files and cards
+    if (eventType === 'm.room.message' && 'data' in content) {
+      const data = JSON.parse(content.data) as CardMessageContent['data'];
```

Comment:
We are going to be doing a lot of JSON parsing here since we will process every aibot streaming (replace) message. Not sure how best to optimize.

Reply:
Updated here [51b7aed](https://github.com/cardstack/boxel/pull/2549/commits/51b7aed90344f3471cf9d0231c4939cada39b28b)